source/adam.cpp

```diff
@@ -33,6 +32,7 @@
 
 #ifndef NDEBUG
 
+#include <functional>
```

Comment:
This includes needs to be outside the conditional block. I belive we use std::function<> even for non-debug builds